Tests/NIOTests/EventLoopFutureTest.swift

```diff
@@ -1023,6 +1023,28 @@ class EventLoopFutureTest : XCTestCase {
             XCTAssert(type(of: error) == EventLoopFutureTestError.self)
         }
     }
+    
+    func testPromiseCompletedWithResultantFuture() throws {
+        let group = EmbeddedEventLoop()
+        let loop = group.next()
+        
+        let success: Result<Int, Error> = .success(42)
+        let failure: Result<Int, Error> = .failure(EventLoopFutureTestError.example)
+
+        let future = (success: loop.makeResultantFuture(success),
+                      failure: loop.makeResultantFuture(failure))
+        
+        let promised = (success: loop.makePromise(of: Int.self),
+                        failure: loop.makePromise(of: Int.self))
+        
+        promised.success.completeWith(future.success)
+        promised.failure.completeWith(future.failure)
```

Comment:
This is quite convoluted: should we just `.wait` for the result and use the regular `XCTAssert` macros on the returned value?